packages/web/src/components/PopupModal/PopupModal.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState } from 'react';
+import {
+  Button,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  useDisclosure,
+} from '@chakra-ui/react';
+
+interface PopupModalProps {
+  openModalText: String;
+  header: String;
+  body: String;
+  openButtonVariant?: 'link' | 'outline' | (string & {}) | 'ghost' | 'solid' | 'unstyled';
+  onConfirm?: () => Promise<void>;
+  succussMessage?: string;
+  errorMessage?: string;
+}
+
+export const PopUpModal: React.FC<PopupModalProps> = ({
+  openModalText,
+  header,
+  openButtonVariant,
+  body,
+  onConfirm,
+  succussMessage,
+  errorMessage,
+}) => {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const [modalHeader, setModalHeader] = useState(header);
+  const [bodyText, setBodyText] = useState(body);
+  const [isLoading, setIsLoading] = useState(false);
+  const [confirmButtonVisible, setConfirmButtonVisible] = useState(true);
+
```

Comment:
Moved the function from the UI into here, and also I added `else` for when there is no success message, the modal should probably close. And if it's an error we can use the error message as the text

```suggestion

  const onConfirmClick = async () => {
    try {
      setIsLoading(true);
      await onConfirm();
      
      if (succussMessage) {
        setModalHeader('Success');
        setBodyText(succussMessage);
      } else {
        onClose();
      }
    } catch (err) {
      setModalHeader('Uh oh');
      setBodyText(errorMessage ?? err.message);
    }
    
    setIsLoading(false);
    setConfirmButtonVisible(false);
  };

```